nss-c31-unusual-prestige/nutshell
# Av user registration

# Description

The user can now register and then log in using their registered email and username. The code will check the login credentials against the users stored in the API, and log the user in if the credentials are correct, storing the necessary user information in session storage for retrieval whenever its required.

The log in section is replaced by a welcome page displaying the logged in user's name, and a log out button. Logging out clear's the storage and returns the user to the login.

Chat is responsive to which user is currently logged in. The current user's name is appended to the front of the message in the chat, and only the user's messages will have edit buttons beside them (a user can't edit another user's message).

Applies to issue # 7

## Type of change: Feature

Please delete options that are not relevant.


# Testing Instructions

- git fetch --all
- git checkout av-user-registration
- make sure this is in your main.js:
   import apiManager from "./apiManager.js";
   import newsFeedDOM from "./newsFeed/newsDomManage.js"
   import newsForm from "./newsFeed/newsForm.js"
   import newsEventHandler from "./newsFeed/newsEventHandler.js"
   import homepage from "./Homepage/homePage.js"
   import messageHandler from "./chat/messages.js"
   // newsForm.newsForm()
   // newsFeedDOM.childArticleContainer()
   // newsFeedDOM.listAllNews()

   homepage.homePage()
   messageHandler.createChat();
- run grunt.
- register a new user.
- log in (make sure you're putting in the right user name and email before telling me it doesn't work).
- Type a message.
- Confirm that the message has your user's name.
- Log out.
- Try creating a new user and logging in with that account.
- See that you can only edit the messages that you post with that account, and not the ones you made 
  previously under another name.

# Checklist:
